src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -86,23 +87,42 @@ void CGaussianKernel::set_width(float64_t w)
 {
 	require(w>0, "width ({}) must be positive",w);
 	m_log_width = std::log(w / 2.0) / 2.0;
+	m_eigen_log_width.value() = m_log_width;
+}
+
+auto CGaussianKernel::kernel_function(int32_t idx_a, int32_t idx_b)
+{
+	// this could be written as Eigen::Matrix<float64_t, n_differentiable_params, 1>;
+	m_eigen_log_width.value() = m_log_width;
+
+	// this could be written as 
+	// eigen_log_width.derivatives() = EigenScalar::Unit(n_differentiable_params, i);
+	// where i is the idx of the adjoint
+	m_eigen_log_width.derivatives() = Eigen::VectorXd::Unit(1,0);
```

Comment:
This is the derivative of the log width wrt the log width itself?

Reply:
this the derivative of the kernel function wrt to the log with, df/dw. It gives the same result as the manual derivative that was in the loop before in get_parameter_gradient

Reply:
Then kernel function should be equal to the log width (for the derivative to be one), no?

Reply:
what do you mean? The derivative is set to one, so that when you apply the chain rule, you get it wrt to this parameter, i.e. you initialise the dependent variable. Like shown here https://eigen.tuxfamily.org/bz_attachmentbase/attachment.cgi?id=395. The same happens in stan gradient function: https://github.com/stan-dev/math/blob/b42294a57318a27d2b6723b96a0b620485ba83e0/stan/math/rev/core/grad.hpp#L39